service/frontend/redirection_interceptor.go

```diff
@@ -162,32 +162,37 @@ var _ grpc.UnaryServerInterceptor = (*RedirectionInterceptor)(nil).Intercept
 
 func (i *RedirectionInterceptor) Intercept(
 	ctx context.Context,
-	req interface{},
+	req any,
 	info *grpc.UnaryServerInfo,
 	handler grpc.UnaryHandler,
-) (_ interface{}, retError error) {
+) (_ any, retError error) {
 	defer log.CapturePanic(i.logger, &retError)
 
+	if _, isWorkflowHandler := info.Server.(*WorkflowHandler); !isWorkflowHandler {
+		return handler(ctx, req)
+	}
```

Comment:
This is actual fix.

Reply:
Is this necessary? All requests will have this extra check.

Reply:
I believe this is very cheap check. The other option is to compare strings: first return value from `SplitMethodName` with hardcoded string `temporal.api.workflowservice.v1.WorkflowService` but I believe checking type is better and faster.

Reply:
Don't forget health check. I see tons of `unknown operation: Check` warning logs.

Reply:
I don't have a strong opinion between this type check and string check either. but I agree we should do one since the redirection logic only applies to workflowservice